cpp/src/arrow/compute/exec/test_util.cc

```diff
@@ -459,5 +460,44 @@ void PrintTo(const Declaration& decl, std::ostream* os) {
   *os << "}";
 }
 
+std::shared_ptr<Table> MakeRandomTable(TableGenerationProperties properties) {
+  int total_columns = properties.num_columns + 2;
+  std::vector<std::shared_ptr<Array>> columns;
+  columns.reserve(total_columns);
+  arrow::FieldVector field_vector = arrow::FieldVector();
+  field_vector.reserve(total_columns);
+
+  field_vector.push_back(std::make_shared<Field>("time", int64()));
+  field_vector.push_back(std::make_shared<Field>("id", int32()));
+
+  int num_rows = 0;
+  std::vector<int64_t> time_column;
+  std::vector<int32_t> id_column;
+  for (int time = properties.start; time <= properties.end;
+         time += properties.time_frequency) {
+    for (int id = 0; id < properties.num_ids; id++) {
+      time_column.push_back(time);
+      id_column.push_back(id);
+      num_rows += 1;
+    }
+  }
+  std::shared_ptr<Array> time_array;
+  ArrayFromVector<Int64Type, int64_t>(int64(), time_column, &time_array);
+  columns.push_back(time_array);
+  std::shared_ptr<Array> id_array;
+  ArrayFromVector<Int32Type, int32_t>(int32(), id_column, &id_array);
+  columns.push_back(id_array);
```

Comment:
It would be slightly more efficient to use `Int64Builder` and `Int32Builder` instead of creating a vector first and then copying the vector into an array.